writer.go

```diff
@@ -1232,20 +1235,29 @@ func (c *writerColumn) flushFilterPages() error {
 		index:              int16(c.bufferIndex),
 	}
 
-	rbuf, pool := getBufioReader(nil, 1024)
+	var pageReader io.Reader = c.pageBuffer
+	if _, err = c.pageBuffer.Seek(0, io.SeekStart); err != nil {
+		return err
+	}
+
+	if _, ok := pageReader.(*os.File); ok {
```

Comment:
I think this is only useful in the case that pageReader is an os.File? If I'm right, this wrapping should perhaps be baked into the fileBufferPool implementation? It might benefit when reading/writing in other places too.

Reply:
I don't know if we can make this assumption as there could be custom implementations of the page buffer which don't use `*os.File` but still benefit from buffering small reads.

If we want to disable buffering we should have an opt-out mechanism rather than opt-in as it's usually the desirable behavior. But unless we have a benchmark showing the benefits I would suggest we avoid early optimizing and keep the code simple.

Reply:
My thinking is that if a custom page buffer implementation benefits from buffering small reads, it should internally use a bufio wrapper.